internal/app/client/conformance_client.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2023 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/url"
+
+	"connectrpc.com/conformance/internal/app"
+	"connectrpc.com/conformance/internal/gen/proto/connect/connectrpc/conformance/v1alpha1/conformancev1alpha1connect"
+	v1alpha1 "connectrpc.com/conformance/internal/gen/proto/go/connectrpc/conformance/v1alpha1"
+	"connectrpc.com/connect"
+)
+
+type conformanceClientWrapper struct {
+	client conformancev1alpha1connect.ConformanceServiceClient
+}
+
+func (w *conformanceClientWrapper) Invoke(
+	ctx context.Context,
+	req *v1alpha1.ClientCompatRequest,
+) (*v1alpha1.ClientCompatResponse, error) {
+	switch req.Method {
+	case "Unary":
+		if len(req.RequestMessages) != 1 {
+			return nil, errors.New("unary calls must specify exactly one request message")
+		}
+		resp, err := w.unary(ctx, req)
+		if err != nil {
+			return nil, err
+		}
+		return resp, nil
+	case "ServerStream":
+		if len(req.RequestMessages) != 1 {
+			return nil, errors.New("server streaming calls must specify exactly one request message")
+		}
+		resp, err := w.serverStream(ctx, req)
+		if err != nil {
+			return nil, err
```

Comment:
Per comment added in default case below: since we've actually invoked the method at this point, we should try to wrap `err` into `ClientCompatResponse.response.error`.